src/scripts/main.js

```diff
@@ -1,3 +1,271 @@
 'use strict';
 
-// write your code here
+// Constants
+const button = document.querySelector('.button');
+const startMessage = document.querySelector('.message-start');
+const scoreText = document.querySelector('.game-score');
+const field = document.querySelector('.game-field');
+const winMessage = document.querySelector('.message-win');
+const loseMessage = document.querySelector('.message-lose');
+
+let board;
+let score = 0;
+const cellsInRow = 4;
+
+const filterZero = (row) => {
+  return row.filter(num => num !== 0);
+};
+
+const compareFields = (prevBoard, boards) => {
+  return JSON.stringify(prevBoard) !== JSON.stringify(boards);
+};
+
+const slide = (row) => {
+  const newRow = filterZero(row);
+
+  for (let i = 0; i < newRow.length - 1; i++) {
+    if (newRow[i] === newRow[i + 1]) {
+      newRow[i] *= 2;
+      newRow.splice(i + 1, 1);
+      score += newRow[i];
+    }
+  }
+
+  while (newRow.length < cellsInRow) {
+    newRow.push(0);
+  }
+
+  return newRow;
+};
+
+const hasEmptyCell = () => {
+  for (let r = 0; r < cellsInRow; r++) {
+    for (let c = 0; c < cellsInRow; c++) {
+      if (board[r][c] === 0) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+};
+
+const renderField = () => {
+  for (let r = 0; r < cellsInRow; r++) {
+    for (let c = 0; c < cellsInRow; c++) {
+      field.rows[r].cells[c].className = '';
+
+      field.rows[r].cells[c].classList.add(
+        'field-cell',
+        `field-cell--${board[r][c]}`
+      );
+      field.rows[r].cells[c].textContent = board[r][c] || '';
+    }
+  }
+};
+
+const spawnRandomCell = () => {
+  if (!hasEmptyCell()) {
+    return;
+  }
+
+  let found = false;
+
+  while (!found) {
+    const r = Math.floor(Math.random() * cellsInRow);
+    const c = Math.floor(Math.random() * cellsInRow);
+
+    if (board[r][c] === 0) {
+      board[r][c] = 2;
+      renderField();
+
+      found = true;
+    }
+  }
+};
+
+const transponseField = (currentField) => {
+  let transponsedBoard = currentField;
+
+  transponsedBoard = transponsedBoard[0].map(
+    (_, colIndex) => transponsedBoard.map(
+      row => row[colIndex]
+    )
+  );
+
+  return transponsedBoard;
+};
+
+const moveLeft = () => {
+  const prevBoard = JSON.parse(JSON.stringify(board));
+
+  for (let r = 0; r < cellsInRow; r++) {
+    let row = board[r];
+
+    row = slide(row);
+    board[r] = row;
+  };
+
+  if (compareFields(prevBoard, board)) {
+    spawnRandomCell();
+  }
+
+  renderField();
```

Comment:
we can move all this logic from move functions to the end of the switch, it will simplify and remove the redundant copy logic in move functions